Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h

```diff
@@ -97,25 +101,55 @@ bool is_outward_oriented(const PolygonMesh& pmesh,
   VPMap vpmap = choose_param(get_param(np, vertex_point),
                              get_const_property_map(vertex_point, pmesh));
   //Kernel
-  typedef typename GetGeomTraits<PolygonMesh, NamedParameters>::type Kernel;
-
-  internal::Compare_vertex_points_xyz_3<typename Kernel::Less_xyz_3, VPMap >
-    less_xyz(vpmap);
+  typedef typename GetGeomTraits<PolygonMesh, NamedParameters>::type GT;
+  GT gt = choose_param(get_param(np, geom_traits), GT());
 
+  //find the vertex with maximal z coordinate
   typename boost::graph_traits<PolygonMesh>::vertex_iterator vbegin, vend;
   cpp11::tie(vbegin, vend) = vertices(pmesh);
-  typename boost::graph_traits<PolygonMesh>::vertex_iterator v_min
-    = std::min_element(vbegin, vend, less_xyz);
-
-  const typename Kernel::Vector_3&
-    normal_v_min = compute_vertex_normal(*v_min, pmesh, np);
-
-  return normal_v_min[0] < 0 || (
-            normal_v_min[0] == 0 && (
-              normal_v_min[1] < 0  ||
-              ( normal_v_min[1]==0  && normal_v_min[2] < 0 )
-            )
-         );
+
+  internal::Compare_vertex_points_z_3<GT, VPMap> less_z(vpmap, gt);
+  typename boost::graph_traits<PolygonMesh>::vertex_iterator v_max_it
+    = std::max_element(vbegin, vend, less_z);
+  typename boost::graph_traits<PolygonMesh>::vertex_descriptor v_max = *v_max_it;
```

Comment:
I think something should be done for empty meshes: either add a predication `!empty`, or return `true` for empty meshes.